djhtml/__main__.py

```diff
@@ -51,12 +57,31 @@ def main():
             _error(e)
             continue
 
+        # Guess tabwidth
+        if not options.tabwidth:
+            diff = 0
+            probabilities = [0] * 9
+            for line in source.split("\n"):
+                count = 0
+                for char in line:
+                    if char == " ":
+                        count += 1
+                    elif char == "\t":
+                        count += 4
```

Comment:
Doesn't this assume a tabwidth of 4 already?

Besides, in Vim parlance, you mean "shiftwidth": https://vimdoc.sourceforge.net/htmldoc/options.html#'shiftwidth' . Not to be confused with `tabstop` which is the multiplier at which a "tab"-keypress stops.

Reply:
Yes, we assume a _tabwidth_ of 4, and we are guessing the _shiftwidth_